radix-engine/src/updates/cuttlefish.rs

```diff
@@ -160,3 +174,86 @@ fn generate_cuttlefish_transaction_validation_updates() -> StateUpdates {
         ),
     )
 }
+
+fn generate_cuttlefish_metadata_fix<S: SubstateDatabase + ?Sized>(db: &S) -> StateUpdates {
+    struct MetadataUpdates {
+        pub name: Option<MetadataValue>,
+        pub description: Option<MetadataValue>,
+        pub icon_url: Option<MetadataValue>,
+    }
+
+    impl MetadataUpdates {
+        pub fn into_map(self) -> IndexMap<String, MetadataValue> {
+            [
+                self.name.map(|value| ("name", value)),
+                self.description.map(|value| ("description", value)),
+                self.icon_url.map(|value| ("icon_url", value)),
+            ]
+            .into_iter()
+            .flatten()
+            .map(|(k, v)| (k.to_owned(), v))
+            .collect()
+        }
+    }
+
+    let reader = SystemDatabaseReader::new(db);
+
+    // The metadata entries that we would like to update and the values that we would like to update
+    // them to be.
+    let metadata_updates = indexmap! {
+        XRD => MetadataUpdates {
+            name: None,
+            description: None,
+            icon_url: Some(MetadataValue::Url(UncheckedUrl("https://assets.radixdlt.com/icons/icon-xrd.png".into())))
+        },
+        PACKAGE_OF_DIRECT_CALLER_RESOURCE => MetadataUpdates {
+            name: Some(MetadataValue::String("Package Caller Resource".into())),
+            description: Some(MetadataValue::String("Badges generated automatically by the Radix system to represent the authority of the package for a direct caller. These badges cease to exist at the end of their transaction.".into())),
+            icon_url: Some(MetadataValue::Url(UncheckedUrl("https://assets.radixdlt.com/icons/icon-package_of_direct_caller_resource.png".into())))
+        },
+        GLOBAL_CALLER_RESOURCE => MetadataUpdates {
+            name: Some(MetadataValue::String("Global Caller Resource".into())),
+            description: Some(MetadataValue::String("Badges generated automatically by the Radix system to represent the authority of a global caller. These badges cease to exist at the end of their transaction.".into())),
+            icon_url: Some(MetadataValue::Url(UncheckedUrl("https://assets.radixdlt.com/icons/icon-global_caller_resource.png".into())))
+        },
+        SECP256K1_SIGNATURE_RESOURCE => MetadataUpdates {
+            name: Some(MetadataValue::String("ECDSA Secp256k1 Signature Resource".into())),
+            description: Some(MetadataValue::String("Badges generated automatically by the Radix system to represent ECDSA Secp256k1 signatures applied to transactions. These badges cease to exist at the end of their transaction.".into())),
+            icon_url: Some(MetadataValue::Url(UncheckedUrl("https://assets.radixdlt.com/icons/icon-ecdsa_secp256k1_signature_resource.png".into())))
+        },
+        ED25519_SIGNATURE_RESOURCE => MetadataUpdates {
+            name: Some(MetadataValue::String("EdDSA Ed25519 Resource".into())),
```

Comment:
This is inconsistent with the above. Should probably be `EdDSA Ed25519 Signature Resource`*